docs/src/main/asciidoc/websockets-next-reference.adoc

```diff
@@ -185,7 +185,8 @@ public class MyWebSocket {
 
 ==== Request context
 
-If an endpoint is annotated with `@RequestScoped`, or with a security annotation (such as `@RolesAllowed`), or depends directly or indirectly on a `@RequestScoped` bean, or on a bean annotated with a security annotation, then each WebSocket endpoint callback method execution is associated with a new _request context_.
+If an endpoint is annotated with `@RequestScoped`, or with a security annotation (such as `@RolesAllowed`) that does not secure an HTTP upgrade,
```

Comment:
```suggestion
If an endpoint is annotated with `@RequestScoped`, or a security annotation such as `@RolesAllowed` that does not secure an HTTP upgrade,
```

Reply:
It may be worth splitting this sentence into something like: `Each WebSocket endpoint callback method execution is associated with a new _request context_ in the following cases:` and list them, as otherwise this sentence looks quite complex, Rolfe would definitely prefer to split it :-)

Reply:
I went with the suggestion to create a list, but didn't exactly go word by word. Please have a look and let's discuss it in a new comment thread (linked to changed text, so that I know what to change), thanks